skopt/searchcv.py

```diff
@@ -268,43 +268,17 @@ class BayesSearchCV(BaseSearchCV):
 
     """
 
-    def __init__(self, estimator, search_spaces=None, optimizer_kwargs=None,
+    def __init__(self, estimator, search_spaces, optimizer_kwargs=None,
```

Comment:
Something that is probably out of scope for this PR: I realize now is that `optimizer_kwargs` are the same for all subspaces. However, one might want these to be different for different subspaces. An example would be that I want optimizer to try a few initial points, that I suspect already to be good for some subspace. This could be done by specifying kwargs for optimizer; However, in case of multiple subspaces this is not possible as kwargs for those are the same. Similarly, one might want to use decision trees based optimizers for subspaces with large number of iterations and models with fast fitting. 
So kwargs could be specified per subspace, like iterations number. For now one can simply have a separate BayesSearchCv per subspace. 
Edit: maybe lets leave this out, in order to avoid everlasting PR's :)

Reply:
I think this could be done in further PR too, I'd like to propose some documentation with further possible improvements.

Reply:
@iaroslav-ai can you make an issue for this so we don't lose the idea?